src/Features/Core/Portable/CodeLens/DisplayInfoProvider.cs

```diff
@@ -0,0 +1,346 @@
+        private static Tuple<string, string, string, string, string, TextSpan> GetReferenceText(Document document, int position)
+        {
+            SyntaxToken token = FindTokenAtPosition(document, position);
+
+            // get the full line of source text on the line that contains this position
+            Task<SourceText> docText = document.GetTextAsync();
+            SourceText text = docText.Result;
+            // get the actual span of text for the line containing reference
+            TextLine textLine = text.Lines.GetLineFromPosition(position);
+            // turn the span from document relative to line relative
+            int spanStart = token.Span.Start - textLine.Span.Start;
+            string line = textLine.ToString();
+
+            string beforeLine1 = string.Empty;
+            if (textLine.LineNumber > 0)
+            {
+                TextLine beforeTextLine = text.Lines[textLine.LineNumber - 1];
+                beforeLine1 = beforeTextLine.ToString();
+            }
+
+            string beforeLine2 = string.Empty;
+            if (textLine.LineNumber - 1 > 0)
+            {
+                TextLine beforeTextLine = text.Lines[textLine.LineNumber - 2];
+                beforeLine2 = beforeTextLine.ToString();
+            }
+
+            string afterLine1 = string.Empty;
+            if (textLine.LineNumber < text.Lines.Count - 1)
+            {
+                TextLine afterTextLine = text.Lines[textLine.LineNumber + 1];
+                afterLine1 = afterTextLine.ToString();
+            }
+
+            string afterLine2 = string.Empty;
+            if (textLine.LineNumber + 1 < text.Lines.Count - 1)
+            {
+                TextLine afterTextLine = text.Lines[textLine.LineNumber + 2];
+                afterLine2 = afterTextLine.ToString();
+            }
+
+            return new Tuple<string, string, string, string, string, TextSpan>(line.TrimEnd(),
+                                                               beforeLine1.TrimEnd(),
+                                                               beforeLine2.TrimEnd(),
+                                                               afterLine1.TrimEnd(),
+                                                               afterLine2.TrimEnd(),
+                                                               new TextSpan(spanStart, token.Span.Length));
+        }
+
+        /// <summary>
+        /// find the token at a given postion in the document
+        /// </summary>
+        /// <param name="document">the document containing the reference</param>
+        /// <param name="position">the position of the reference</param>
+        /// <returns>the syntax token at the specified location</returns>
+        private static SyntaxToken FindTokenAtPosition(Document document, int position)
+        {
+            Task<SyntaxNode> taskGetSyntaxRoot = document.GetSyntaxRootAsync();
```

Comment:
I'm sure this will be a common thing to watch out for. Codelens in VS repo did not have access to internal roslyn helpers, so we may not need to bring in all of its helpers.